Source/EasySynth/Private/CameraRig/CameraRigRosInterface.cpp

```diff
@@ -0,0 +1,265 @@
+
+#include "CameraRig/CameraRigRosInterface.h"
+
+#include "CineCameraComponent.h"
+#include "DesktopPlatformModule.h"
+#include "IDesktopPlatform.h"
+#include "JsonObjectConverter.h"
+#include "Kismet/KismetMathLibrary.h"
+#include "Misc/FileHelper.h"
+#include "Serialization/JsonReader.h"
+#include "Serialization/JsonSerializer.h"
+
+#include "CameraRig/CameraRigData.h"
+#include "CoordinateSystemConverter.h"
+#include "EasySynth.h"
+
+
+#define LOCTEXT_NAMESPACE "FCameraRigRosInterface"
+
+FReply FCameraRigRosInterface::OnImportCameraRigClicked()
+{
+	UE_LOG(LogEasySynth, Log, TEXT("%s"), *FString(__FUNCTION__))
+
+	// Get desktop platform
+	void* ParentWindowPtr = FSlateApplication::Get().GetActiveTopLevelWindow()->GetNativeWindow()->GetOSWindowHandle();
+	IDesktopPlatform* DesktopPlatform = FDesktopPlatformModule::Get();
+	if (DesktopPlatform == nullptr)
+	{
+		UE_LOG(LogEasySynth, Error, TEXT("%s: Could not get the desktop platform"), *FString(__FUNCTION__))
+		return FReply::Handled();
+	}
+
+	// Display file open dialog
+	TArray<FString> OutFilenames;
+	const bool IsFileSelected = DesktopPlatform->OpenFileDialog(
+		ParentWindowPtr,
+		TEXT("Import camera rig"),
+		TEXT(""),
+		TEXT(""),
+		TEXT("Camera Rig ROS JSON (*.json)|*.json"),
+		EFileDialogFlags::None,
+		OutFilenames);
+	if (!IsFileSelected)
+	{
+		return FReply::Handled();
+	}
+
+	// Helper method for displaying error messages
+	auto DisplayError = [](const FText& Message) {
+		const FText MessageBoxTitle = LOCTEXT("InvalidJsonMessageBoxTitle", "Failed to load camera rig");
+		UE_LOG(LogEasySynth, Warning, TEXT("%s: %s"), *FString(__FUNCTION__), *Message.ToString())
+		FMessageDialog::Open(EAppMsgType::Ok, Message, &MessageBoxTitle);
+	};
+
+
+	// Read the selected file
+	FString FileContent;
+	if (!FFileHelper::LoadFileToString(FileContent, *OutFilenames[0]))
+	{
+		const FText ErrorMessage = LOCTEXT("CannotReadFileError", "Failed to open ROS JSON file");
+		DisplayError(ErrorMessage);
+		return FReply::Handled();
+	}
+
+	// Parse JSON content into equivalent structure
+	FRosJsonContent RosJsonContent;
+	if (!FJsonObjectConverter::JsonObjectStringToUStruct(FileContent, &RosJsonContent, 0, 0))
+	{
+		const FText ErrorMessage = LOCTEXT("FileContentError", "Invalid ROS JSON file content");
+		DisplayError(ErrorMessage);
+		return FReply::Handled();
+	}
+
+	// Aggregated camera rig data
+	FCameraRigData CameraRigData;
+
+	for (auto& Element : RosJsonContent.cameras)
+	{
+		const FString& CameraName = Element.Key;
+		const FRosJsonCamera& RosJsonCamera = Element.Value;
+
+		FCameraRigData::FCameraData CameraData;
+		CameraData.CameraName = CameraName;
+
+		// Get the sensor size
+		if (RosJsonCamera.sensor_size.Num() != 2)
+		{
+			const FText ErrorMessage = LOCTEXT("InvalidSensorError", "Expected 2 values for the sensor size");
+			DisplayError(ErrorMessage);
+			return FReply::Handled();
+		}
+		CameraData.SensorSize.X = RosJsonCamera.sensor_size[0];
```

Comment:
Can we use sensor size to modify Camera aspect ratio? I noticed I have to do it manually for all cameras after import

Reply:
I mean Sensor Width/Height params, if aspect ratio is not correct rendered images are wrong size
![image](https://user-images.githubusercontent.com/57099007/183875749-a0486a83-f009-4731-99f3-342335bd4356.png)

Reply:
Good catch, don't know how it didn't surface earlier.
Actually, the camera sensor width should be adjusted before rendering to match the requested output image aspect ratio.